### PR TITLE
In-memory mode support

### PR DESCRIPTION
**Pull request by a5anka**
Implement an in-memory mode which will skip the database layer completely when used. This means we will be doing all of the following tasks in memory.
- Persistent message storing
- Durable queue information storing
- Durable exchange information storing
- Durable binding information storing

Implement an in-memory mode to skip the database access layer.

Set of null objects were introduced to skip the database access layer when in-mmory mode is enabled.

N/A

- Support in-memory mode

COnfiguration related info added to https://github.com/ballerina-platform/ballerina-message-broker/blob/master/docs/user/configure.md

N/A

N/A

## Automation tests
 - Unit tests - 77% 
 - Integration tests - in-progress

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: docs/user/configure.md [hunk 1]
```diff
@@ -9,24 +9,32 @@ The main configuration file of the broker can be located at ${message.broker.hom
...
+|-----------------------------| ---------------------------------------|-----------------------------------------------|
+| enableInMemoryMode          | false                                  | When the in-memory mode is enabled all message and context data including durable entities are held in memory. lease note that this can increase memory usage of the broker. Therefore it is advised to allocate more memory in the JVM which run the broker. | 
+| datasource:url              | jdbc:h2:./database/MB_DB               | Database URL.                                 |
+| database:user               | wso2carbon                             | Database username                             |
+| database:password           | wso2carbon                             | Database password.                            |
+
 ### Broker-core configurations
 
-These configurations are defined under the namespace `wso2.broker`. 
+These configurations are defined under the namespace `ballerina.broke.core`.
```
Comment (MaryamZi) [inline docs/user/configure.md]: broke --> broker

line 14 too :)

Comment (a5anka) [reply docs/user/configure.md]: Fixed.

File: modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/MemBackedQueueHandlerFactory.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Override
+    public QueueHandler createDurableQueueHandler(String queueName, boolean autoDelete) {
+        return createNonDurableQueueHandler(queueName, autoDelete);
+    }
+
+    @Override
+    public QueueHandler createNonDurableQueueHandler(String queueName, boolean autoDelete) {
+        Queue queue = new MemQueueImpl(queueName, nonDurableQueueMaxDepth, autoDelete);
+        return new QueueHandler(queue, metricManager);    }
```
Comment (MaryamZi) [inline modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/MemBackedQueueHandlerFactory.java]: formatting issue?

Comment (a5anka) [reply modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/MemBackedQueueHandlerFactory.java]: Fixed

File: modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/store/NullMessageStore.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    @Override
+    public void fillMessageData(QueueBuffer queueBuffer, Message message) {
+        // Do nothing
+    }
+
+    @Override
+    public Collection<Message> readAllMessagesForQueue(String queueName) {
+        return EMPTY_COLLECTION;
```
Comment (Asitha) [inline modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/store/NullMessageStore.java]: Shall we use ```Collections.emptyList();```?

**Pull request merged**